CONTRIBUTING.md

```diff
@@ -1,21 +1,132 @@
-## Guidelines
+## Contributing Guidelines
 
-- Add one change per one commit.
+### Development environment
+Mimesis does not have any need for third-party tools, but we use a lot of tools in development stage,
+which you should install on your system if you want to contribute.
 
-- Always comment your code (only in English!).
+We use `Pipenv` for managing development packages:
+```
+~ pip install pipenv
+~ pipenv install --dev
+```
 
-- Check your spelling and grammar.
+### Code Style
+1. Every contributor must follow the [PEP8](https://www.python.org/dev/peps/pep-0008/) code style.
+2. Always use single-quoted strings, unless a single-quote occurs within the string.
+3. Always run tests, because tests check code style too (`pytest-flake8`).
 
-- Check code with [pycodestyle](https://github.com/PyCQA/pycodestyle) or with any other similar tool.
+### Annotating
+We are using optional static typing (`mypy`) and this means that every contributor should
+annotate his functions and methods (not variables, because Python 3.5 does not support annotation of variables).
```

Comment:
I suggest to rephrase this part as: "We use optional static typing (`mypy`). Every function and method should be annotated."

Also link to the `mypy` docs would be awesome.